src/pages/iou/request/step/IOURequestStepTag.js

```diff
@@ -64,11 +67,19 @@ function IOURequestStepTag({
      * @param {String} selectedTag.searchText
      */
     const updateTag = (selectedTag) => {
-        if (selectedTag.searchText === tag) {
-            IOU.resetMoneyRequestTag_temporaryForRefactor(transactionID);
-        } else {
-            IOU.setMoneyRequestTag_temporaryForRefactor(transactionID, selectedTag.searchText);
+        const isSelectedTag = selectedTag.searchText === tag;
+        const updatedTag = !isSelectedTag ? selectedTag.searchText : '';
+        if (isSplitBill) {
```

Comment:
Coming from https://github.com/Expensify/App/issues/35836. This uses the SPLIT_TRANSACTION_DRAFT Onyx key, whereas we use the TRANSACTION_DRAFT key when creating a request, so we end up updating the wrong key here.